Lesson17/Lesson17/gokce/script.js

```diff
@@ -0,0 +1,217 @@
+        maxNumber=findMaxNumberInArray[j];
+    }   
+}
+console.log(maxNumber);
+
+// 5. Write a loop that finds the minimum number in an array of numbers.
+console.log("Exercise 5");
+const findMinNumberInArray = [1, 10, 23, 42, 0, 12, 7, 0];
+let minNumber=findMinNumberInArray[0];
+for (let m=0; m<findMinNumberInArray.length; m++){
+    if (findMinNumberInArray[m]<minNumber){
+        minNumber = findMinNumberInArray[m];
+    }
+}
+console.log(minNumber);
+
+// 6. Write a loop that reverses a given string.
+console.log("Exercise 6");
+const string = 'Lorem ipsum dolor sit amet consectetur adipisicing elit.';
+let reversedString;
+for (let y=string.length; y>=0; y--){
+    reversedString += string [y];
+}
+console.log(reversedString);
+
+// 7. Write a while loop that counts from 10 to 1 and prints out each number.
+console.log("Exercise 7");
+let i=10;
+while(i>=1){
+    console.log(i);
+    i--;
+}
+
+// 8. Write a while loop that generates random numbers between 1 and 10 until a number greater than 8 is generated, and then prints out the total number of iterations.
+console.log("Exercise 8");
+let iterationsCount = 0;
+let chosenNumber = 0;
+while (chosenNumber<=8){
+    chosenNumber = Math.floor(Math.random()*10)+1;
+    console.log(chosenNumber);
+    iterationsCount++;
+}
+console.log("Total iterations "+ iterationsCount);
+
+
+// 9. Write a do-while loop that counts from 1 to 10 and prints out each number.
+console.log("Exercise 9");
+let c=1;
+do{
+    console.log(c);
+    c++;
+}
+while (c<11);
+
+// 10. Write a for...in loop that prints out the keys and values of an object.
+console.log("Exercise 10");
+const person = { name: 'John', surname: 'Doe', age: 30, city: 'New York' };
+for (let info in person){
+    console.log(info + ": " + person[info]);
+}
+ 
+
+// 11. Write a for...in loop that sums the values of an object.
+console.log("Exercise 11");
+const obj = { a: 10, b: 20, c: 30, d: 23 };
+let sum = 0;
+for (let numbers in obj){
+    sum += obj[numbers];
+    console.log ("the sum of the numbers are" + sum);
+}
+
+
+// Excercises below are additional (as requested during the lesson)
+
+// 12. Sum of Random Numbers: Write a do-while loop that generates random numbers between 1 and 5 and adds them together until the sum is greater than 20. Print the total number of iterations and the final sum.
+console.log("Exercise 12");
+let totalResult = 0;
+let totalIterations = 0;
+do{
+ let randomNum = Math.floor(Math.random()*5)+1;
+ totalResult += randomNum;
+ console.log(totalResult);
+ totalIterations++;
+}
+while(totalResult<=20);
+console.log("iterations : " + totalIterations);
+console.log("total result : " + totalResult );
+
+
+// 13. Count Even Numbers: Write a for loop that generates and checks 100 random numbers between 1 and 50. Count how many even numbers are generated and print the count at the end.
+console.log("Exercise 13");
+let sumIteration = 0;
+let evenNum =0;
+do{
+let fullNum = Math.floor(Math.random()*50)+1; 
```

Comment:
I think it would be better to name this variable 'randomNumber' as this is what the value is (the code generates a random number and assigns it to the variable)